test/fw/ptl/lib/pbs_testlib.py

```diff
@@ -13034,6 +13044,47 @@ def __init__(self, username=None, attrs={}, jobname=None):
         self.set_variable_list(self.username)
         self.set_sleep_time(100)
 
+    def set_attributes(self, a={}):
+        """
+        set attributes and custom attributes on this object.
+        custom attributes are used when converting attributes to CLI.
+        In case of Cray platform if 'Resource_List.vntype' is set
+        already then remove it and add vntype value to each chunk of a
+        select statement.
+
+        :param a: Attribute dictionary
+        :type a: Dictionary
+        """
+        if isinstance(a, list):
+            a = OrderedDict(a)
+
+        self.attributes = OrderedDict(self.dflt_attributes.items() +
+                                      self.attributes.items() + a.items())
+
+        if self.platform == 'cray' or self.platform == 'craysim':
+            s = 'Resource_List.select' in a.keys()
+            v = 'Resource_List.vntype' in self.custom_attrs.keys()
+            if s and v:
+                ra = []
+                del self.custom_attrs['Resource_List.vntype']
+                r = a['Resource_List.select'].split('+')
+                for i in r:
+                    select = PbsTypeSelect(i)
+                    novntype = 'vntype' not in select.resources.keys()
+                    nohost = 'host' not in select.resources.keys()
+                    novnode = 'vnode' not in select.resources.keys()
+                    if novntype and nohost and novnode:
+                        i = i + ":vntype=cray_compute"
+                    ra.append(i)
+                select_str = ''
+                for l in ra:
+                    select_str = select_str + "+" + l
+                select_str = select_str[1:]
+                a['Resource_List.select'] = select_str
```

Comment:
Since rebuilding the select resource with vntype is the same as the \_\_init\_\_() function, you could factor it out into its own function.  That way it'd share code and make the two parent functions cleaner.